src/garage/envs/bullet/bullet_env.py

```diff
@@ -1,65 +1,69 @@
+    'ThrowerBulletEnv', 'Walker2DBulletEnv', 'InvertedPendulumBulletEnv',
+    'InvertedDoublePendulumBulletEnv', 'InvertedPendulumSwingupBulletEnv',
+    'HalfCheetahBulletEnv', 'AntBulletEnv', 'HopperBulletEnv',
+    'HumanoidBulletEnv', 'HumanoidFlagrunBulletEnv',
+    'HumanoidFlagrunHarderBulletEnv'
+]
 
-class BulletEnv(gym.Wrapper):
+
+class BulletEnv(GymEnv):
     """Binding for py_bullet environments."""
 
-    def __init__(self, env=None, env_name='', is_image=False):
-        """Returns a Garage wrapper class for bullet-based gym.Env.
+    def __new__(cls, *args, **kwargs):
+        """Overwrite GymEnv's __new__ logic.
 
         Args:
-            env (gym.wrappers.time_limit): A gym.wrappers.time_limit.TimeLimit
-                object wrapping a gym.Env created via gym.make().
-            env_name (str): If the env_name is speficied, a gym environment
-                with that name will be created. If such an environment does not
-                exist, a `gym.error` is thrown.
-            is_image (bool): True if observations contain pixel values,
-                false otherwise. Setting this to true converts a gym.Spaces.Box
-                obs space to an akro.Image and normalizes pixel values.
+            args: positional arguments
+            kwargs: keyword arguments
+
+        Returns:
+             garage.envs.bullet.BulletEnv
 
         """
-        if not env:
-            # 'RacecarZedBulletEnv-v0' environment enables rendering by
-            # default, while pybullet allows only one GUI connection at a time.
-            # Setting renders to False avoids potential error when multiple
-            # of these envs are tested at the same time.
-            if env_name == 'RacecarZedBulletEnv-v0':
-                env = gym.make(env_name, renders=False)
-            else:
-                env = gym.make(env_name)
-
-        # Needed for deserialization
-        self._env = env
-        self._env_name = env_name
-
-        super().__init__(env)
-        self.action_space = akro.from_gym(self.env.action_space)
-        self.observation_space = akro.from_gym(self.env.observation_space,
-                                               is_image=is_image)
-        self._spec = EnvSpec(action_space=self.action_space,
-                             observation_space=self.observation_space)
-
-    @property
-    def spec(self):
-        """Return the environment specification.
-
-        This property needs to exist, since it's defined as a property in
-        gym.Wrapper in a way that makes it difficult to overwrite.
+        return super(BulletEnv, cls).__new__(cls)
 
-        Returns:
-            garage.envs.env_spec.EnvSpec: The envionrment specification.
+    def __init__(self, env, is_image=False, max_episode_length=None):
+        """Returns a wrapper class for bullet-based gym.Env.
+
+        Args:
+            env (gym.wrappers.time_limit or str): A gym.TimeLimit
```

Comment:
```suggestion
            env (gym.Env or str): A gym.TimeLimit
```

This docstring is incorrect -- your API can accept any `gym.Env` or a `str` representing an `env_id`. There's no reason it needs to be a `TimeLimit` wrapper.